### PR TITLE
Update price buy native based price when there is no execution price

### LINKED ISSUE #1541: Market price is disabled when clear out the field
1. Open Limit order page
2. Select a token pair, an amount to trade
3. Clear out the Price field

**AR**: Market price is disabled. It remains disabled until a user inputs a value into the field

**ER**: Market price button is enabled when a value in the price field differs from the value in the Price input (even if it 0)

### PR DESCRIPTION
**Pull request by shoom3301**
# Summary

Fixes #1541

https://user-images.githubusercontent.com/7122625/204782310-1110ec76-f06b-4ca9-8085-a493335d363c.mov

1. Refresh native-based price every 10 sec (`https://barn.api.cow.fi/goerli/api/v1/token/0xxxxnative_price`)
2. When the execution price is not set (the rate input is empty) we change text on the button from "Market price" to "Update price"
3. By clicking on "Update price" we set the native-based price as the active price

### CODE REVIEW

**Comment by nenadV91** [comment]
I've noticed that now if we change the currencies, the price is not recalculated!

**Comment by shoom3301** [comment]
> I've noticed that now if we change the currencies, the price is not recalculated!

Thanks @nenadV91 ! Fixed

**Comment by elena-zh** [comment]
Hey @shoom3301 , when I revert a toke pair several times, the price is changed to 1
change a token pair

Another issue is that 'Update price' disabled button is displayed when the price is fetched.  Then it is changed to Market price. Maybe we shouldn't show it at all?
disabled

Also is this OK to see disabled Update price button when there are no amounts in the fields? 
image
Actually, I see that sometimes the button is enabled
image

**Comment by anxolin** [comment]
> Another issue is that 'Update price' disabled button is displayed when the price is fetched. Then it is changed to Market price. Maybe we shouldn't show it at all?

I think is fine as it is. I don't see any problem showing the disabled field as we do in this PR. Let me know if there's any big reason to change it

**Comment by anxolin** [comment]
> It works like before with "Market price", We disable "Update price" when entered value is the same as the native-based price

I like this behaviour. If I see it disabled, I know my limit price, matches the price of the button.  All good from my side

File: src/cow-react/modules/limitOrders/containers/RateInput/index.tsx [hunk 1]
```diff
@@ -54,16 +66,22 @@ export function RateInput() {
...
+  }, [activeRate, executionRate, isLoadingExecutionRate, initialRate])
 
   return (
     <styledEl.Wrapper>
       <styledEl.Header>
         <HeadingText currency={primaryCurrency} rateImpact={rateImpact} />
 
         <styledEl.MarketPriceButton disabled={isDisabledMPrice} onClick={handleSetMarketPrice}>
-          <span>Market price</span>
+          <span>{executionRate ? 'Market price' : 'Update price'}</span>
```
Comment (anxolin) [inline src/cow-react/modules/limitOrders/containers/RateInput/index.tsx:84]: small question with label https://cowservices.slack.com/archives/C0361CDG8GP/p1669823385058719?thread_ts=1669818427.372219&cid=C0361CDG8GP

File: src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts [hunk 2]
```diff
@@ -42,39 +45,48 @@ async function requestPriceForCurrency(chainId: number | undefined, currency: Cu
...
-  }, [outputPrice, inputPrice])
-
-  // To avoid loading state blinking
+  const [updateTimestamp, setUpdateTimestamp] = useState(Date.now())
+
+  const price = useAsyncMemo(
+    () => {
+      setIsLoading(true)
+
+      return Promise.all([
```
Comment (anxolin) [inline src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts:59]: would it make sense to create a function that would do this 2 request, and then then, so this part gets simplified? 

sth like 
```js
const price = useAsyncMemo(
    () => {
      setIsLoading(true)
      return getMarketPrice(chainId, inputCurrency outputCurrency)
              .finally(() => setLoading(false))
     })
```

**Review by anxolin — APPROVED** [review]

**Comment by elena-zh** [comment]
Hey @shoom3301 , changes LGTM!
The only one issue I'd like to mention is that it would be great to disable 'Update price' button when the 2nd token is not selected:
image

File: src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts (hunk 2 above)
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/hooks/useGetInitialPrice.ts:59]: Thanks! Done

File: src/cow-react/modules/limitOrders/containers/RateInput/index.tsx (hunk 1 above)
Comment (shoom3301) [reply src/cow-react/modules/limitOrders/containers/RateInput/index.tsx:84]: Answered. Removed this

**Comment by shoom3301** [comment]
@elena-zh thanks! Fixed

**Review by elena-zh — APPROVED** [review]
LGTM!

**Pull request merged**